sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -108,11 +108,15 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
   {
     std::shared_lock<std::shared_timed_mutex> Lock(MGraphLock);
 
-    Command *NewCmd = static_cast<Command *>(NewEvent->getCommand());
+    // enqueueCommand() func below may throw an exception, so use shared_ptr
+    // here to avoid memory leak
+    std::shared_ptr<Command> NewCmd(
```

Comment:
Is it exclusively `shared_ptr` which is needed here? Will `unique_ptr` fulfill the needs?

Reply:
Yes, `shared_ptr` doesn't need at all, replaced with `unique_ptr`.